test/cliffordsynthesis/test_synthesis.cpp

```diff
@@ -236,4 +245,103 @@ TEST_P(SynthesisTest, TwoQubitGatesMinimalGatesMaxSAT) {
             test.expectedMinimalGatesAtMinimalTwoQubitGates);
 }
 
+TEST(SynthesisTest, DestabilizerSanityCheck) {
+  Tableau             initialTableau;
+  Tableau             targetTableau;
+  Configuration       config;
+  CliffordSynthesizer synthesizer;
+  Results             results;
+
+  initialTableau = Tableau(2);
+  targetTableau  = Tableau("[+IX, +ZI]");
+
+  config.target    = TargetMetric::Gates;
+  config.useMaxSAT = true;
+
+  synthesizer = CliffordSynthesizer(initialTableau, targetTableau);
+  synthesizer.synthesize(config);
+  results = synthesizer.getResults();
+
+  EXPECT_EQ(results.getGates(), 3);
+
+  initialTableau = Tableau(2, true);
+  targetTableau  = Tableau("[+IX, +ZI]", "[+ZZ, -XX]");
+
+  config.target    = TargetMetric::Gates;
+  config.useMaxSAT = true;
+
+  synthesizer = CliffordSynthesizer(initialTableau, targetTableau);
+  synthesizer.synthesize(config);
+  results = synthesizer.getResults();
+
+  EXPECT_EQ(results.getGates(), 4);
+}
+
+TEST_P(SynthesisTest, TestDestabilizerGates) {
+  if (!initialTableauWithDestabilizer.getTableau().empty()) {
+    std::cout << "Testing with destabilizer" << std::endl;
+    config.target    = TargetMetric::Gates;
+    config.useMaxSAT = true;
+
+    synthesizer.synthesize(config);
+    synthesizerWithDestabilizer.synthesize(config);
+    results                 = synthesizer.getResults();
+    resultsWithDestabilizer = synthesizerWithDestabilizer.getResults();
+
+    EXPECT_GE(resultsWithDestabilizer.getGates(), results.getGates());
+  } else {
+    std::cout << "Testing without destabilizer" << std::endl;
+    config.target    = TargetMetric::Gates;
+    config.useMaxSAT = true;
+
+    synthesizer.synthesize(config);
+    results = synthesizer.getResults();
+  }
```

Comment:
The initial tableau is never empty, as it is always populated in the setup routine.
So the whole `else` branch can be eliminated here (it is redundant anyways, as it is already covered by the checks above).
This alleviates the need for the `getTableau` method.

Reply:
I don't think that that is true. `initialTableauWithDestabilizer` is not always populated, specifically when the tableaus are extracted from a circuit and an additional initial tableau is given. If the initial tableau doesn't include the destabilizers then the `initialTableauWithDestabilizer` can't really be populated as the choice of destabilizers can be arbitrary. I guess one could just pick the diagonal tableau but for a test framework I don't think that is a good idea as that makes the results somewhat arbitrary.

Reply:
Agreed. In that case, leave it as it is.